libretro-common/include/retro_unittest.h

```diff
@@ -0,0 +1,102 @@
+/* Copyright  (C) 2010-2015 The RetroArch team
+ *
+ * ---------------------------------------------------------------------------------------
+ * The following license statement only applies to this file (retro_stat.h).
+ * ---------------------------------------------------------------------------------------
+ *
+ * Permission is hereby granted, free of charge,
+ * to any person obtaining a copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,
+ * and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+#ifndef _RUNIT_H
+#define _RUNIT_H
+/* This is heavily inspired by Minunit. */
+#include <stdbool.h>
+#include <stdio.h>
+
+#define RUNIT_TEST(method_name) static void method_name()
+#define RUNIT_TEST_SUITE(suite_name) static void suite_name()
+
+#define RUNIT_MESSAGE_LEN 1024
+#define _RUNIT_PRINTF(fmt,args...) fprintf(stderr, fmt, ##args)
+#define _RUNIT_FLUSH() fflush(stderr)
+#define _RUNIT_PRINT_FAIL(msg) snprintf(runit_last_message, RUNIT_MESSAGE_LEN, "[FAIL] %s (in %s:%d %s)", msg, __FILE__, __LINE__, __func__)
+#ifdef RUNIT_SHOW_PASS
+ #define _RUNIT_PRINT_PASS(msg) _RUNIT_PRINTF("[PASS] %s\n", msg)
+#else
+ #define _RUNIT_PRINT_PASS(msg)
+#endif
+static void (*runit_setup)(void) = NULL;
+static void (*runit_teardown)(void) = NULL;
+
+static int
+   runit_run      = 0,
+   runit_assert   = 0,
+   runit_fail     = 0,
+   runit_status   = 0;
```

Comment:
Please either use `static int` for every variable or merge them in a single line.